src/core/server/http/lifecycle/on_pre_response.ts

```diff
@@ -24,9 +24,16 @@ import { Logger } from '../../logging';
 import { HapiResponseAdapter, KibanaRequest, ResponseHeaders } from '../router';
 
 enum ResultType {
+  render = 'render',
   next = 'next',
 }
 
+interface Render {
+  type: ResultType.render;
+  body: string;
+  statusCode?: number;
```

Comment:
shouldn't we allow `headers?: ResponseHeaders;` as well? to be compatible with `next`? we might have to change render interface to `render({body, headers} : { body: string, headers? : Headers})`

Reply:
I was considering it, but didn't want to add too many features I didn't use in the PR itself. But if you're ok with it, I'll happily add `headers` as well 👍

Reply:
While working on this, I discovered that since we're actually creating a brand new response-object, it doesn't contain any of the original headers that was about to be sent to the client. Should we copy over all the headers from the original response object to the newly created one, or should it only use the headers specified by the user when calling `render`?

Reply:
I've implemented it like that for now (i.e. that it copies over all the headers from the original response while allowing you to override whichever header you want).